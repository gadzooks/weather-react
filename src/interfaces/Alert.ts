interface Alert {
  id: string;
  event: string;
  headline: string;
  description: string;
  ends: string;
  endsEpoch: number;
  link: string;
}

export default Alert;
