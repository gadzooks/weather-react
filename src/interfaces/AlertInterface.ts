interface AlertInterface {
    event: string;
    headline: string;
    description: string;
    ends: string;
    id: string;
    link: string;
}

export default AlertInterface;
