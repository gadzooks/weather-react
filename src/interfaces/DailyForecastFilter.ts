export interface DailyForecastFilter {
    date?: string;
    tempmax?: number;
    tempmin?: number;
    precip?: number;
    precipprob?: number;
}
