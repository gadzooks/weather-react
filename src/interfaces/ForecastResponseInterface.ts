import { LocationInterface } from './LocationInterface';
import { RegionInterface } from './RegionInterface';
import { DailyForecastInterface } from './DailyForecastInterface';

export interface RegionById {
    [key: string]: RegionInterface;
}

export interface RegionsById {
  byId: RegionById;
  allIds: string[];
}

export interface LocationById {
  [key: string]: LocationInterface;
}

export interface ForecastById {
  [key: string]: DailyForecastInterface[];
}

export interface ForecastsById {
  byId: ForecastById;
}

export interface LocationsById {
  byId: LocationById;
  allIds: string[];
}

export interface ForecastResponse {
  dates: string[];
  regions: RegionsById;
  locations: LocationsById;
  forecasts: ForecastsById;
}
export interface ForeacastDates {
  dates: string[];
  parsedDates: Date[];
  weekends: boolean[];
}

export const DefaultForecastDates: ForeacastDates = {
  dates: [],
  parsedDates: [],
  weekends: [],
};

export interface ForecastResponseStatus {
  isLoaded?: boolean;
  error: Error | null;
  forecast: ForecastResponse | null;
  forecastDates: ForeacastDates;
}

export const DefaultForecastResponseStatus: ForecastResponseStatus = {
  isLoaded: false,
  error: null,
  forecast: null,
  forecastDates: DefaultForecastDates,
};

// dates: [
//     "2021-04-17T07:00:00.000+00:00",
//     "2021-04-18T07:00:00.000+00:00",
//     "2021-04-19T07:00:00.000+00:00",
//     "2021-04-20T07:00:00.000+00:00",
// ],
// regions: {
//     byId: {
//     central_cascades: {
//         id: "central_cascades",
//         description: "Central Cascades",
//         regionId: "central_cascades",
//         locations: ["renton", "yakima"],
//     },
//     western_wa: {
//         id: "western_wa",
//         description: "Western WA",
//         regionId: "western_wa",
//         locations: ["seattle", "bellevue"],
//     },
//     },
//     allIds: ["central_cascades", "western_wa"],
// },
// locations: {
//     byId: {
//     renton: {
//         id: "renton",
//         description: "Renton",
//         regionId: "central_cascades",
//     },
//     yakima: {
//         id: "yakima",
//         description: "Yakima",
//         regionId: "central_cascades",
//     },
// forecasts: {
//     byId: {
//       renton: [
//         {
//           time: "2021-04-17T07:00:00.000+00:00",
//           summary: "Clear conditions throughout the day.",
//           icon: "day-hail",
//           precipProbability: 0.0,
//           temperature: 60.9,
//           apparentTemperature: 60.9,
//           dewPoint: 40.1,
//           visibility: 120,
//           cloudCover: 1,
//           temperatureHigh: 74.9,
//           temperatureLow: 47.3,
//           sunsetTime: 1618714951,
//           sunriseTime: 1618665359,
//           precipAmount: 0.0,
//         },
