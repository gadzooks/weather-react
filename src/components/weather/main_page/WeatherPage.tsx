/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { DailyForecastFilter } from '../../../interfaces/DailyForecastFilter';
import {
  ForecastResponseStatus,
} from '../../../interfaces/ForecastResponseInterface';
import { MatchedAreas } from '../../../interfaces/MatchedAreas';
import SummaryTable, { SummaryTableProps } from '../forecast_summary/SummaryTable';
// import LocationDetails from '../location_details/LocationDetails';

export function WeatherForecastSubPage(props: SummaryTableProps) {
  // const { forecastResponse } = props;

  return (
    <>
      <div id='top' />
      <SummaryTable {...props} />

      {/* {forecastResponse && (
        <LocationDetails
          matchedAreas={props.matchedAreas}
          forecastsByName={forecastResponse.forecasts.byId}
          isWeekend={props.forecastDates.weekends}
          dates={props.forecastDates.parsedDates}
        />
      )} */}
    </>
  );
}

export interface WeatherPageArgs {
  appState: ForecastResponseStatus,
  matchedAreas: MatchedAreas,
  dailyForecastFilter: DailyForecastFilter,
}

function WeatherPage(props: WeatherPageArgs) {
  const { appState } = props;
  const args:SummaryTableProps = {
    ...props,
    forecastDates: appState.forecastDates,
    forecastResponse: appState.forecast,
  };

  return (
    <>
      {appState?.error && (
        <div>
          Error:
          {appState.error?.message}
        </div>
      )}
      {!appState.isLoaded && <div>Loading...</div>}
      {appState.forecast && <WeatherForecastSubPage {...args} />}
    </>
  );
}

export default WeatherPage;
