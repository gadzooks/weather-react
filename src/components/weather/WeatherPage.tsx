import { format } from "fecha";
import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { ForecastResponse } from "../../interfaces/ForecastResponseInterface";
import SearchableTableHook from "./forecast_summary/SearchableTableHook";

export function isWeekend(dates: (Date|null)[]) :boolean[] {
    const weekends: boolean[] = [];
    if(!dates) {
        return weekends;
    }

    dates.map((d) => {
        if (d) {
            const dayOfWeek = format(d, 'ddd').toUpperCase();
            weekends.push(dayOfWeek === 'SUN' || dayOfWeek === 'SAT')
        } else {
            weekends.push(false);
        }
    })

    return weekends; 
}

interface AppState {
  isLoaded: boolean,
  error: Error|null,
  forecast: ForecastResponse|null,
}

const WeatherPage = (props: AppState) => {

  const [appState, setAppState] = useState(props);
  const params = useParams();
  const dataSource = params.dataSource || 'useLocal';

  useEffect(() => {
    const WEATHER_API = process.env.REACT_APP_WEATHER_API;

    const url = `${WEATHER_API}/forecasts/${dataSource}`;
    console.log(`getting weather from ${url}`);
    fetch(`${url}`, { mode: 'cors' })
      .then(res => res.json())
      .then(
        (result) => {
          setAppState({
            isLoaded: true,
            forecast: result.data,
            error: null,
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          setAppState({
            isLoaded: true,
            error: error,
            forecast: null,
          });
        }
      )
  }, [])

  return(
    <>
    { appState.error && <div>Error: {appState.error.message}</div> }
    { !appState.isLoaded && <div>Loading...</div> }
    { appState.forecast && <Page {...appState.forecast} /> }
    </>
  );
}

export default WeatherPage;

export function Page(props: ForecastResponse) {
  return (
    <>
      <div id='top' />
      <SearchableTableHook {...props} />
    </>
  )

}