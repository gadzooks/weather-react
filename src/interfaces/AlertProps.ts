import { AlertsById } from './ForecastResponseInterface';

interface AlertProps {
  allAlerts: string[] | null;
  alertsById: AlertsById | null;
  foundAlerts: boolean;
}

export default AlertProps;
