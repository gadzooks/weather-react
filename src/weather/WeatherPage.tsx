import { parse, format } from "fecha";
import React from "react";
import { ForecastResponse } from "../interfaces/ForecastResponseInterface";
import SearchableTableHook from "./forecast_summary/SearchableTableHook";
import LocationDetails from "./location_details/LocationDetails";

function isWeekend(dates: (Date|null)[]) :boolean[] {
    const weekends: boolean[] = [];
    if(!dates) {
        return weekends;
    }

    dates.map((d) => {
        if (d) {
            const dayOfWeek = format(d, 'ddd').toUpperCase();
            weekends.push(dayOfWeek === 'SUN' || dayOfWeek === 'SAT')
        } else {
            weekends.push(false);
        }
    })

    return weekends; 
}

interface AppState {
  isLoaded: boolean,
  error: Error|null,
  forecast: ForecastResponse|null,
}

class WeatherPage extends React.Component<AppState, AppState> {
  constructor(props: AppState) {
    super(props);
    this.state = {
      isLoaded: false,
      error: null,
      forecast: null,
    }
  }

  componentDidMount() {
    // const result = mockWeatherForecastNormalized();
    // this.setState({
    // })
    const WEATHER_API = process.env.REACT_APP_WEATHER_API;

    //TODO: get params from url and decide url vs mock
    fetch(`${WEATHER_API}/forecasts/mock`, {mode:'cors'})
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            forecast: result.data,
            error: null,
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )
  }

  render() {
    const { error, isLoaded, forecast } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div>Loading...</div>;
    } else if (forecast != null) {
      return <Page {...forecast} />
    }
  }
}

function Page(props: ForecastResponse) {
      const parsedDates = props.dates.map((d) => parse(d, 'YYYY-MM-DD'));
      const weekends = isWeekend(parsedDates);
      const args = {
        ...props,
        weekends,
        parsedDates,
      };

      return (
        <>
          <SearchableTableHook isWeekend={weekends} {...args}/>
          <LocationDetails forecastsByName={props.forecasts.byId} isWeekend={weekends} locations={props.locations.allIds} dates={parsedDates} />
        </>
      )

}

export default WeatherPage;