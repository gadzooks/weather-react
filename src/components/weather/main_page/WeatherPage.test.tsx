import React from 'react';
import renderer from 'react-test-renderer';
import mockWeatherForecastNormalized from '../../../api/mockData';
import { ForecastResponse } from '../../../interfaces/ForecastResponseInterface';
import { Page } from './WeatherPage';

it('renders a snapshot', () => {
    const mockData = mockWeatherForecastNormalized().data as unknown as ForecastResponse;
    const tree = renderer.create(
        <Page {...mockData} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});