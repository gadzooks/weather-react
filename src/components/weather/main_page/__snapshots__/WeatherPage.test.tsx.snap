// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
Array [
  <div
    id="top"
  />,
  <table
    className="MuiTable-root table table-sm weather-forecast-summary css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root table-heading css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Location
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SAT APR 17
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SUN APR 18
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          MON APR 19
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          TUE APR 20
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          WED APR 21
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          THU APR 22
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          FRI APR 23
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SAT APR 24
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SUN APR 25
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          MON APR 26
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          TUE APR 27
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          WED APR 28
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          THU APR 29
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          FRI APR 30
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          SAT MAY 01
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root region-details css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          colSpan={16}
        >
          Cities
        </td>
      </tr>
      <tr
        className="MuiTableRow-root weather-cell css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell css-1ex1afd-MuiTableCell-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="#seattle"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            SEATTLE CITY
          </a>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-cloudy day-cloudy "
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-cloudy day-cloudy "
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with morning rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with rain in the morning and afternoon."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Cloudy skies throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Cloudy skies throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-10"
            title="Partly cloudy throughout the day."
          />
        </td>
      </tr>
      <tr
        className="MuiTableRow-root weather-cell css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell css-1ex1afd-MuiTableCell-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="#bonney_lake"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            BONNEY LAKE
          </a>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-sunny day-sunny "
            title="Clear conditions throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with rain in the morning and afternoon."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-day-cloudy day-cloudy "
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Partly cloudy throughout the day with a chance of rain."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-rain rain "
            title="Cloudy skies throughout the day with rain in the morning and afternoon."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Partly cloudy throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center   css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-50"
            title="Cloudy skies throughout the day."
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium weather-cell center  weekend  css-1ex1afd-MuiTableCell-root"
        >
          <i
            className="wi weather-icon wi-cloudy cloudy sunshine-10"
            title="Clearing in the afternoon."
          />
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
